### PR TITLE
tests: QoL improvements for running tests locally

### PR DESCRIPTION
**Pull request by sprt**
This is a series of patches to make running tests outside of GHA more pleasant.

Fixes: #7454

### CODE REVIEW

File: tests/integration/kubernetes/gha-run.sh [hunk 1]
```diff
@@ -121,6 +121,11 @@ function cleanup() {
     echo "Gather information about the nodes and pods before cleaning up the node"
     get_nodes_and_pods_info "yes"
 
+    if [ "${platform}" = "aks" ]; then
+        delete_cluster
+        return
+    fi
```
Comment (fidencio) [inline tests/integration/kubernetes/gha-run.sh:131]: Please, allow kata-deploy to be uninstalled from aks clusters, so we can know whether this actually working or not.

If you want to add a short-circuit to go directly to the cliuster deletion, I'm fine, but in the CI runs I'd prefer not changing the folow.

**Comment by fidencio** [comment]
@sprt, **very nice series!!!**

I've left one comment, once that's adressed I'll approve it.

File: tests/integration/kubernetes/gha-run.sh (hunk 1 above)
Comment (fidencio) [reply tests/integration/kubernetes/gha-run.sh:131]: Ouch, it seems this is needed as without it we've been leaving pods behind.

I am approving it, and later on we can improve it if needed.

**Review by fidencio — APPROVED** [review]
lgtm, thanks @sprt!

**Review by jepio — APPROVED** [review]
looks ok

File: tests/integration/kubernetes/gha-run.sh (hunk 1 above)
Comment (sprt) [reply tests/integration/kubernetes/gha-run.sh:131]: > Please, allow kata-deploy to be uninstalled from aks clusters, so we can know whether this actually working or not.

This makes sense, hadn't thought about this.

> Ouch, it seems this is needed as without it we've been leaving pods behind.

Don't understand this one actually. Can you explain? Aren't we fine if we delete the cluster in the end?

Comment (fidencio) [reply tests/integration/kubernetes/gha-run.sh:131]: It seems we're failing in a cleanup step and we're never triggering the cluster delete part. :-/

**Pull request merged**